### PR TITLE
Save the set of running scripts when they change, as well as at shutdown

### PR DESCRIPTION
**Pull request by jherico**
## Testing

* Start, stop and restart the application while running some set of scripts, verify scripts are preserved over the restart.
* Add a new script through the Running Scripts dialog.  Wait 10 second and then crash the application through the developer menu.  Restart the application and verify the newly added script is still loaded.

### CODE REVIEW

**Comment by jherico** [comment]
https://app.asana.com/0/26225263936266/110570190234757

**Comment by hifi-gustavo** [comment]
Build console output for commit: c5b574664672aaebe6ecce27a71a9c23c4ceecbc available at: https://gist.github.com/8b633db59b875a42d3f8ffe88f9e06c3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7628.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7628.dmg

**Comment by kevinmthomas-carpool** [comment]
## QA Test
**PR Build:** 7628
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** sandbox, playa, cellscience, forum
**Smoke Test Results:** successful

                                  | playa   | cellscience | forum   
----------------------------------|---------|-------------|--------
Render Rate:                      |61       |62           |64           
Present Rate:                     |60       |60           |60           
Simrate:                          |61       |61           |63            
Render Stats Num Buffers:         |3233     |2752         |4508              
Render Stats gpu::Buffer Memory:  |73.2Mb   |30.0Mb       |51.8Mb              
Render Stats Num Textures:        |421      |186          |776              
Render Stats gpu::Texture Memory: |932.9Mb  |272.4Mb      |851.9Mb              
Render Stats Triangles:           |835k     |566k         |1048k            
Render Stats Drawcalls:           |45240    |19911        |15407             
Render Stats Items:               |214      |806          |362             

**sandbox ‘createtestblocks.js’ Result:** Failure and blocks behaved in an unexpected state as if there was a constant physical pressure against them causing them to move rapidly and unexpectedly without the contact of an avatar.
**sandbox Import FBX Result:** Successful test.
**sandbox ‘harmonicOscillator.js’ Result:** Successful test.
**sandbox Echo Server Audio Result:** Failure as voice was robotic and another user was able to confirm such.
**sandbox ‘reverbTest.js’ Result:** Failure as the UI did not load and was unable to adjust properties however the reverb settings did work properly.
**Additional Test Results Requested:** (none requested)
**Successful Logout/Crash (If  A Crash Occurs, Attach ‘hifi-log.txt’):** Successful logout of Interface.

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -731,11 +731,18 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer) :
...
+    connect(scriptSaveTimer, &QTimer::timeout, [] {
+        DependencyManager::get<ScriptEngines>()->saveScripts();
+    });
+    scriptSaveTimer->setSingleShot(true);
+    connect(scriptEngines, &ScriptEngines::scriptCountChanged, scriptEngines, [this, scriptSaveTimer] {
         auto scriptEngines = DependencyManager::get<ScriptEngines>();
         if (scriptEngines->getRunningScripts().isEmpty()) {
             getMyAvatar()->clearScriptableSettings();
         }
+        scriptSaveTimer->start(SCRIPT_SAVE_COUNTDOWN_INTERVAL_MS);
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: this feels weirdly spaghetti to me... can we make this "saving of scripts" part of the `ScriptEngines` class. 

And why exactly does `MyAvatarThe fact that the avatar s

Comment (ZappoMan) [inline interface/src/Application.cpp]: looking at `clearScriptableSettings()` it appears this relates to `MyAvatar.motorVelocity` which appears to be used in the sample `move.js` I'm not entire sure that code has been tested recently and I wonder if it even works.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5a5583d3a5b56eb2bb251047503d4b87f864ec7a available at: https://gist.github.com/8b633db59b875a42d3f8ffe88f9e06c3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7628.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7628.dmg

**Comment by kevinmthomas-carpool** [comment]
RETEST successful - Physics and server audio and reverb test successful.

**Pull request merged**